routers/api/v1/repo/repo.go

```diff
@@ -34,31 +34,30 @@ func Search(ctx *context.APIContext) {
 		OwnerID:  ctx.QueryInt64("uid"),
 		PageSize: convert.ToCorrectPageSize(ctx.QueryInt("limit")),
 	}
-	if ctx.User != nil && ctx.User.ID == opts.OwnerID {
-		opts.Searcher = ctx.User
-	}
 
-	// Check visibility.
-	if ctx.IsSigned && opts.OwnerID > 0 {
-		if ctx.User.ID == opts.OwnerID {
-			opts.Private = true
-			opts.Collaborate = true
+	if opts.OwnerID > 0 {
+		var repoOwner *models.User
+		if ctx.User != nil && ctx.User.ID == opts.OwnerID {
+			repoOwner = ctx.User
 		} else {
-			u, err := models.GetUserByID(opts.OwnerID)
+			var err error
+			repoOwner, err = models.GetUserByID(opts.OwnerID)
 			if err != nil {
 				ctx.JSON(500, api.SearchError{
```

Comment:
Should it be 404 if user doesn't exist?

Reply:
@ethantkoenig don't thinks so

Reply:
@lafriks IMO, it doesn't make sense to return a 500 if the client provides a bad user ID

Reply:
@ethantkoenig yes but not 404. It should be some other error or just empty repo list

Reply:
@lafriks @ethantkoenig Sounds like a usual 400 Bad Request for me.

Reply:
I didn't change it as it is current behaviour. I'll change it to "bad request".

Reply:
@ethantkoenig @daviian I see problem, this response is defined in api (swagger):
```
	//     Responses:
	//       200: SearchResults
	//       500: SearchError
```

Reply:
I would keep it as it is now and make changes in v2 api. I'll fix only leaking internal information from response. Do you agree?

Reply:
@Morlinest @ethantkoenig as this is documented so let's not introduce breaking changes in API

Reply:
IMO, calling this a breaking change seems like a stretch, but I don't want to block this PR so I'll accept